connect/transforms/src/test/java/org/apache/kafka/connect/transforms/CastTest.java

```diff
@@ -320,6 +324,68 @@ public void castWholeRecordValueSchemalessUnsupportedType() {
         xformValue.apply(new SourceRecord(null, null, "topic", 0, null, Collections.singletonList("foo")));
     }
 
+    @Test
+    public void castLogicalToPrimitive() {
+        List<String> specParts = Arrays.asList(
+            "date_to_int32:int32",  // Cast to underlying representation
+            "timestamp_to_int64:int64",  // Cast to underlying representation
+            "time_to_int64:int64",  // Cast to wider datatype than underlying representation
+            "decimal_to_int32:int32",  // Cast to narrower datatype with data loss
+            "timestamp_to_float64:float64",  // loss of precision casting to double
+            "date_to_string:string",  // cast to string representation
```

Comment:
I think it'd be useful to verify the behavior of casting all of the logical types to strings, just so that we verify the formats (e.g., timestamps, times, and dates should use the ISO 8601 representation) and have some regression tests for the future to help ensure we maintain backward compatibility.

Reply:
Done! I moved the castToString tests to their own testcase.

Looking back at release branches, it seems schema was not piped into castValueToType() until 2.1. Before that, we can not easily determine if we are casting from a logical type. So, I'd say easiest is to go back to 2.1. Otherwise, we will have to backport KAFKA-6684 as well.

Reply:
If that's the case, then we may need a separate PR to go back further than 2.1. I'd say we'd want to correct this issue (which has irked quite a few people) as far back as 1.0.

Reply:
https://github.com/apache/kafka/pull/4820 cherry-picks pretty cleanly onto 1.0. Then, these changes apply cleanly. My vote would be to take both fixes for KAFKA-6684 and this fix back to 1.0. WDYT?

Reply:
Ping @rhauch . WDYT about taking both KAFKA-6684 and this PR as far back as 1.0?